Makefile

```diff
@@ -37,7 +37,7 @@ OUTPUT_DIR := $(abspath .output)
 GO_DIR := $(OUTPUT_DIR)/go
 
 # Base image used for all golang containers
-GOLANG_IMAGE := golang:1.21.5-bullseye
+GOLANG_IMAGE := golang:1.21.5-bookworm
```

Comment:
The `BUILDENV_IMAGE` version below should be bumped so that the `buildenv` image can be rebuilt with this new base image

e.g. `$(TEST_INFRA_REGISTRY)/buildenv:v0.3.1` -> `$(TEST_INFRA_REGISTRY)/buildenv:v0.3.2`

Reply:
do i need to build it manually?

Reply:
No, it's built/published automatically

Reply:
Should i bump the buildenv number in a subsequent PR or this PR?

Reply:
IMO should be done in this PR to make sure the new buildenv passes validation/presubmits before changing the base image

Reply:
An example for how it could fail is if there are some OS packages we use which were available in bullseye but not bookworm